gdnative-core/src/dictionary.rs

```diff
@@ -134,64 +178,219 @@ impl Dictionary {
     /// Modifying the same underlying collection while observing the safety assumptions will
     /// not violate memory safely, but may lead to surprising behavior in the iterator.
     #[inline]
-    pub fn iter(&self) -> Iter {
-        Iter::new(self)
+    pub fn iter(&self) -> IterShared {
+        IterShared::new(self)
     }
 
-    #[doc(hidden)]
+    /// Create a copy of the dictionary.
+    ///
+    /// This creates a new dictionary and is **not** a cheap reference count
+    /// increment.
     #[inline]
-    pub fn sys(&self) -> *const sys::godot_dictionary {
-        &self.0
+    pub fn duplicate(&self) -> Dictionary<Unique> {
+        let d = Dictionary::new();
+        for (k, v) in self {
+            d.insert(&k, &v);
+        }
+        d
     }
+}
 
-    #[doc(hidden)]
+/// Operations allowed on non-shared Dictionaries.
+impl Dictionary<Unique> {
+    /// Creates an empty `Dictionary`.
     #[inline]
-    pub fn from_sys(sys: sys::godot_dictionary) -> Self {
-        Dictionary(sys)
+    pub fn new() -> Self {
+        unsafe {
+            let mut sys = sys::godot_dictionary::default();
+            (get_api().godot_dictionary_new)(&mut sys);
+            Self::from_sys(sys)
+        }
+    }
+
+    /// Put this dictionary under the "shared" access type.
+    #[inline]
+    pub fn into_shared(self) -> Dictionary<Shared> {
+        Dictionary::<Shared> {
+            sys: self.sys,
+            _marker: PhantomData,
+        }
+    }
```

Comment:
`self` is dropped here, but the reference isn't incremented because the constructor is used directly.